lib/ret_web/channels/hub_channel.ex

```diff
@@ -491,19 +498,46 @@ defmodule RetWeb.HubChannel do
     end
   end
 
+  def handle_in("delete_entity_state", %{"nid" => nid, "file_id" => file_id}, socket) do
+    {:ok, hub, account} = authorize(socket, :write_entity_state)
+
+    with {:ok, _} <- Ret.delete_entity(hub.hub_id, nid) do
+      OwnedFile.set_inactive(file_id, account.account_id)
+      :ok
+    else
+      _ ->
+        if account |> can?(pin_objects(hub)) do
+          RoomObject.perform_unpin(hub, nid)
+          OwnedFile.set_inactive(file_id, account.account_id)
+        end
+    end
+
+    broadcast!(socket, "entity_state_deleted", %{
+      "nid" => nid,
+      "creator" => socket.assigns.session_id
+    })
+
+    {:reply, :ok, socket}
+  end
```

Comment:
This handler (and the following one) looks suspicious.

If for some reason the `delete_entity_state` request does not result in the entity state being deleted (and the associated `OwnedFile` becoming inactive), this function should `reply_error` with a reason. The reasons we should catch are:
- If the socket is unauthorized,
- If `delete_entity` fails (e.g. the entity doesn't exist),
- (Maybe?) If the `OwnedFile` doesn't exist

Also, I think we need to handle the `RoomObject` records slightly differently. 

On the client, when we load the legacy room objects, we synthesize messages for them so that the client can act _as if_ they are entity create/update messages. (See `loadLegacyRoomObjects` and `messageForLegacyRoomObject` in the client.)

We would like to add to this flow a step where a client, when it synthesizes this message, also calls `save_entity_state` so that that data is migrated to the new EntityState storage (non destructively).

Therefore, when we receive an `entity_state_deleted` message, there are two (valid) states we should be concerned with:
1. There is both an `EntityState` and a `RoomObject` for this `nid`. We need to delete _both_.
2. There is only an `EntityState` for this `nid`.  We only need to delete the entity state.

Deleting a non-existant RoomObject is probably no trouble, so trying to delete both each time is probably a fine strategy (knowing that many times, the `RoomObject` won't exist).

And, as you have done here -- if there is a `file_id` in the payload, we need to mark the associated `OwnedFile` as `inactive`. To avoid duplicating code, might consider having a single `handle_in("delete_entity_state", %{"nid" => nid } = payload ... `, and if `payload` has a `file_id`, call `set_inactive`. But that's more a matter of style, not correctness.